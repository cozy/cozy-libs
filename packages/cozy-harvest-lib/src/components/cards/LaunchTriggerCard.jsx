import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import Button from 'cozy-ui/transpiled/react/Button'
import Card from 'cozy-ui/transpiled/react/Card'
import { translate } from 'cozy-ui/transpiled/react/I18n'
import Icon from 'cozy-ui/transpiled/react/Icon'
import { Text } from 'cozy-ui/transpiled/react/Text'

import TriggerLauncher from '../TriggerLauncher'

export class LaunchTriggerCard extends PureComponent {
  render() {
    const { className, trigger, t, ...rest } = this.props
    return (
      <Card className={className} {...rest}>
        <TriggerLauncher trigger={trigger}>
          {({ error, launch, running }) => (
            <div>
              {error && (
                <Text className="u-error">{t('card.launchTrigger.error')}</Text>
              )}
              <Button
                label={t('card.launchTrigger.button.label')}
                icon={<Icon focusable="false" icon="sync" spin={running} />}
                className="u-mv-half u-mh-0"
                disabled={running}
                onClick={launch}
                subtle
              />
            </div>
          )}
        </TriggerLauncher>
      </Card>
    )
  }
}

LaunchTriggerCard.propTypes = {
  ...Card.propTypes,
  trigger: PropTypes.object.isRequired,
  t: PropTypes.func.isRequired
}

export default translate()(LaunchTriggerCard)
