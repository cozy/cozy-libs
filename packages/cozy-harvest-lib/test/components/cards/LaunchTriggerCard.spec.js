import React from 'react'
import { shallow } from 'enzyme'

import LaunchTriggerCard from 'components/cards/LaunchTriggerCard'

const fixtures = {
  trigger: {
    _id: 'd861818b62204988bf0bb78c182a9149'
  }
}

const props = {
  trigger: fixtures.trigger,
  t: key => key
}

describe('LaunchTriggerCard', () => {
  it('should render', () => {
    const component = shallow(<LaunchTriggerCard {...props} />).getElement()
    expect(component).toMatchSnapshot()
  })
})
