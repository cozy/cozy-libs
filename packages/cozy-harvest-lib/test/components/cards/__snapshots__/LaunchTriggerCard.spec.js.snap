// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchTriggerCard should render 1`] = `
<LaunchTriggerCard
  trigger={
    Object {
      "_id": "d861818b62204988bf0bb78c182a9149",
    }
  }
/>
`;
